docs/source/schemas/dynamic_columns.rst

```diff
@@ -0,0 +1,140 @@
+Also, key-value pairs in an individual row will be returned in *lexicographically* sorted order of the key.
+
+Dynamic Columns are useful for avoiding KVS-level range scans, especially in key-value services where range scans
+are expensive (like Cassandra - its caching optimisations are rendered ineffectual for range scans). For example,
+for the schema defined above:
+
+1. "Find John's smallest and cheapest todo" can be answered very efficiently.
+   Elements are stored in sorted order, so the very first key-value pair retrieved from the database is actually the
+   correct answer to this query.
+2. "Find all of Tom's todos with size up to M, limited to N results" is easy, as this can be
+   processed with a natural stopping point, as in query 1, and we are iterating in the right order for this query.
+   This can be achieved using the getRowsColumnRange API.
+3. "Find all of Tom's todos with size of 10 to 15, limit to N results" can be performed via a range
+   scan on the dynamic column *values*. This is also readily supported.
+4. "Find all of Tom's largest todos" is somewhat more difficult, as reverse range scans aren't supported by the
+   dynamic columns API. If we were more interested in largest todos as opposed to smallest, we could set the
+   ``ValueByteOrder`` of the ``taskSize`` column component to decreasing.
+5. "Find all of Tom's todos which cost less than 5" will not be efficient; we need to retrieve the entire row to do
+   this, because there is no sorting by sub-units.
+6. "Find all of Tom's todos which have size between 3 and 7, and cost between 5 and 10". This is achievable, though in
+   addition to the values we want to see, we also need to scan through all values with size 4-6, and those with size
+   3 and cost >10, or 7 and cost <5.
+7. "Find all of John's and Jeremy's smallest todos" is easy for a similar reason to 1. We provide an API that allows
+   for this to be done seamlessly as well.
+8. "Find the highest priority todos for everyone whose name begins with J" is NOT easy. While this
+   conceivably can be split into two parts (a prefix scan for J, and then the aforementioned getRowsColumnRange),
+   the range scan for J is potentially costly, as it needs to iterate through every todo for people whose names
+   do indeed begin with J.
+
+Using Dynamic Columns
+---------------------
+
+Storage
+=======
+
+One can use the standard ``put`` method. This can be used to add one or more columns to a row.
+
+.. code:: java
+
+    TodoTable todoTable = TodoSchemaTableFactory.of().getTodoTable(tx);
+
+    todoTable.put(
+            TodoTable.TodoRow.of(person),
+            TodoTable.TodoColumnValue.of(
+                    TodoTable.TodoColumn.of(size1, cost1), description1),
+                    TodoTable.TodoColumn.of(size2, cost2), description2));
+
+Note that if the table already consists of a value for a given row-key and dynamic column values, then this put
```

Comment:
i tend to think of dynamic columns as a list of key-value pairs within a row. in that sense it would be more clear if this sentence said "..for a given row-key and dynamic column key"

Reply:
Yep, agreed. I've added a line in the overview defining some terms (row key, dynamic column key and value). With that terminology, this does read row-key and DC key.